src/client/eppo-client.ts

```diff
@@ -462,22 +462,31 @@ export default class EppoClient {
     actions: BanditActions,
     defaultValue: string,
   ): IAssignmentDetails<string> {
-    const flagEvaluationDetailsBuilder = this.flagEvaluationDetailsBuilder(flagKey);
-    const defaultResult = { variation: defaultValue, action: null };
+    //const flagEvaluationDetailsBuilder = this.flagEvaluationDetailsBuilder(flagKey);
     let variation = defaultValue;
     let action: string | null = null;
+
+    // Initialize with a generic evaluation details. This will mutate as the function progresses.
+    let evaluationDetails: IFlagEvaluationDetails = this.flagEvaluationDetailsBuilder(
+      flagKey,
+    ).buildForNoneResult(
+      'ASSIGNMENT_ERROR',
+      'Unexpected error getting assigned variation for bandit action',
```

Comment:
This will cover the case if we hit an error before getting the updated evaluation details from `getStringAssingmentDetails()`. That method has it's own try catch, so really this just covers the base if we hit an error parsing attributes.